File: lab2/F1.cs
using System;
using System.Threading;

namespace Lab2App {
  [Obsolete("Not used any more", true)]
  public class F1 {

    public static int count = 0; // количество созданных потоков
    String name; // имя потока
    Matrix f1 = new Matrix(0, 0);
    MatrixCalculations matrixCalc = MatrixCalculations.Instance;
    Logger logger = Logger.Instance;

    public F1(String name) {
      this.name = name.ToUpper();
    }

    public void run() {
      Thread.CurrentThread.Name = this.name;
      long start = 0, finish = 0;
      logger.start(
        "  Thread " + Thread.CurrentThread.ManagedThreadId + " named '" + Thread.CurrentThread.Name + "' is running..."
      );
      try {
        // Displaying the thread that is running
        start = Logger.nanoTime();
        f1 = matrixCalc.f1();
        finish = Logger.nanoTime();
      } catch (Exception e) {
        // Throwing an exception
        Console.WriteLine("  Exception is caught: " + e.Message);
      }
      logger.end(
        "  Thread " + Thread.CurrentThread.Name + " is over with data: ",
        "  Thread " + Thread.CurrentThread.Name + " is over with time: " + Logger.getTime(finish - start),
        f1
      );
    }
  }
}

File: lab1/F1.java
package lab1;

public class F1 implements Runnable {

  public static int count = 0; // количество созданных потоков
  String name; // имя потока
  Matrix f1 = new Matrix(0, 0);
  MatrixCalculations matrixCalc = MatrixCalculations.getInstance();

  public F1(String name) {
    this.name = name.toUpperCase();
  }

  // А=В*(МС+MZ)+E*MM-МО
  @Override
  public void run() {
    long start = 0, finish = 0;
    Thread.currentThread().setName(this.name);
    Logger.start(
      "  Thread " + Thread.currentThread().getId() + " named '" + Thread.currentThread().getName() + "' is running..."
    );
    try {
      // Displaying the thread that is running
      start = System.nanoTime();
      f1 = matrixCalc.f1();
      finish = System.nanoTime();
    } catch (Exception e) {
      // Throwing an exception
      System.out.println("  Exception is caught");
    }
    Logger.end(
      "  Thread " + Thread.currentThread().getId() + " named '" + Thread.currentThread().getName() + "' is over with data: ",
      "  Thread " + Thread.currentThread().getId() + " named '" + Thread.currentThread().getName() + "' is over with time: " + Logger.getTime(finish - start),
      f1
    );
  }
}